src/orion/executor/multiprocess_backend.py

```diff
@@ -0,0 +1,209 @@
+
+    def __init__(self, future):
+        self.future = future
+
+    def get(self, timeout=None):
+        return self.future.result(timeout)
+
+    def wait(self, timeout=None):
+        wait([self.future], timeout)
+
+    def ready(self):
+        return self.future.done()
+
+    def successful(self):
+        if not self.future.done():
+            raise ValueError()
+
+        return self.future.exception() is None
+
+
+class ThreadPool:
+    """Custom pool that creates multiple threads instead of processess"""
+
+    def __init__(self, n_workers):
+        self.pool = ThreadPoolExecutor(n_workers)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.pool.shutdown()
+
+    def shutdown(self):
+        self.pool.shutdown()
+
+    def apply_async(self, fun, args, kwds=None):
+        if kwds is None:
+            kwds = dict()
+
+        return _ThreadFuture(self.pool.submit(fun, *args, **kwds))
+
+
+class PoolExecutor(BaseExecutor):
+    """Simple Pool executor.
+
+    Parameters
+    ----------
+
+    n_workers: int
+        Number of worker to spawn
+
+    backend: str
+        Pool backend to use; thread or multiprocess, defaults to multiprocess
+
+    """
+
+    BACKENDS = dict(
+        thread=ThreadPool,
+        threading=ThreadPool,
+        multiprocess=Pool,
+        loky=Pool,
+    )
+
+    def __init__(self, n_workers, backend="multiprocess", **kwargs):
+        self.pool = PoolExecutor.BACKENDS.get(backend, ThreadPool)(n_workers)
+        super().__init__(n_workers, **kwargs)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.pool.shutdown()
+
+    def __del__(self):
+        self.pool.shutdown()
+
+    def __getstate__(self):
+        state = super(PoolExecutor, self).__getstate__()
+        return state
+
+    def __setstate__(self, state):
+        super(PoolExecutor, self).__setstate__(state)
+
+    def submit(self, function, *args, **kwargs) -> AsyncResult:
+        return self._submit_cloudpickle(function, *args, **kwargs)
+
+    def _submit_python(self, function, *args, **kwargs) -> AsyncResult:
+        return _Future(self.pool.apply_async(function, args=args, kwds=kwargs))
+
+    def _submit_cloudpickle(self, function, *args, **kwargs) -> AsyncResult:
+        payload = cloudpickle.dumps((function, args, kwargs))
+        return _Future(self.pool.apply_async(_couldpickle_exec, args=(payload,)), True)
+
+    def wait(self, futures):
+        return [future.get() for future in futures]
+
+    def _find_future_exception(self, future):
+        for _, status in self.futures.items():
+            if id(status.future) == id(future):
+                return status.exception
+
+    def async_get(self, futures, timeout=None):
+        results = []
+        tobe_deleted = []
+
+        for future in futures:
+            if timeout:
+                future.wait(timeout)
```

Comment:
I cannot find the doc for future.wait so I can't say for sure, but I'm concerned the wait could raise an error if the child process crash during future.wait() instead of during future.get().

Reply:
https://docs.python.org/3/library/multiprocessing.html#multiprocessing.pool.AsyncResult.wait

Reply:
There is no information on how this is handled in the doc. :( I would expect that the error is only raised during get(), but it is not specified.

Reply:
well now it is tested anyway with `test_execute_async`